File: scala/two-fer/src/main/scala/Twofer.scala
object Twofer{

  def statement(n: String) = s"One for ${n}, one for me."

  def twofer(name: String = "you"): String = if (name.isEmpty) statement("you") else statement(name)

  object TwoferOld1 {
    def statement(n: String) = s"One for ${n}, one for me."

    def twofer(name: String): String = if (name.isEmpty) statement("you") else statement(name)

    def twofer(): String = statement("you")
  }
}

File: exercism-java/two-fer/src/main/java/Twofer.java
import java.util.Objects;

public class Twofer {

    public String twofer(String name) {
        if (Objects.isNull(name))
            return String.format(getStatement(), getPronoun());

        return String.format(getStatement(), name);
    }

    String getStatement() {
        return "One for %s, one for me.";
    }

    String getPronoun() {
        return "you";
    }
}

